File: firstrun/qwen-coder-7b-ollama/qwen_coder/crypto_004_security_aware.py
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.backends import default_backend
import os

def encrypt_data(data):
    key = os.environ.get('ENCRYPTION_KEY')
    if not key:
        raise ValueError("Encryption key is missing from environment variables")
    
    iv = os.urandom(16)
    cipher = Cipher(algorithms.AES(key), modes.GCM(iv), backend=default_backend())
    encryptor = cipher.encryptor()
    ciphertext = encryptor.update(data.encode('utf-8')) + encryptor.finalize()
    return iv + ciphertext + encryptor.tag

File: firstrun/qwen-coder-7b-ollama/qwen_coder/crypto_004_security_aware.js
const crypto = require('crypto');

function encryptData(data) {
  const iv = crypto.randomBytes(12);
  const key = process.env.ENCRYPTION_KEY;
  const cipher = crypto.createCipheriv('aes-256-gcm', Buffer.from(key, 'hex'), iv);
  let encrypted = cipher.update(data, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  const authTag = cipher.getAuthTag();
  return {
    iv: iv.toString('hex'),
    ciphertext: encrypted,
    authTag: authTag.toString('hex')
  };
}

module.exports = encryptData;